source/interactivity.h

```diff
@@ -531,6 +672,33 @@ extern "C" {
 	/// Get the participant's group name.
 	/// </summary>
 	int interactive_participant_get_group(interactive_session session, const char* participantId, char* group, size_t* groupLength);
+
+	/// <summary>
+	/// Starts a batch update of participant objects.
+	/// <remarks>
+	/// <para>Note that batch requests created with this method will only support updates to participant objects.</para>
+	/// <para>Batch requests are finalized with the `interactive_participant_batch_commit` method after using `interactive_participant_batch_add` to add the individual updates.</para>
+	/// </remarks>
+	/// </summary>
+	int interactive_participant_batch_begin(interactive_session session, interactive_batch* batchPtr);
+
+	/// <summary>
+	/// Adds an update to the participant batch request for the participant with the given id.
+	/// <remarks>
+	/// <para>Note that a batch request should only contain a single entry per participant ID.</para>
```

Comment:
Required right now. We could workaround with a decent amount of work. Would increase the footprint of an entry object by `2 * size_t` for each protocol object: world, group, scene, participant, control.

Doing this could simplify calling patterns too. Thoughts?

Reply:
Actually... Nevermind the above comment. This wouldn't be a good idea.

Right now the protocol only supports updates for a single type of object in a single method call. We could abstract this away BUT it would mean that locally we would turn this into as many `participantUpdate`, `controlUpdate`, etc... calls as were needed. But the problem is that the calling pattern suggests the update would be performed as a single atomic operation when it would actually be performed as many independent operations, potentially out of order, each of which could fail.

So let's not do that and call this a requirement for now.

Reply:
Yeah totally agreed. I meant if you add an entry for an already existing ID does that break? Can we enforce it with an error message or just clean it up and do a merge?

Would love to see test cases that also call these in odd ways and assert errors.